internal/internal_workflow.go

```diff
@@ -532,7 +547,11 @@ func (c *channelImpl) ReceiveAsync(valuePtr interface{}) (ok bool) {
 
 func (c *channelImpl) ReceiveAsyncWithMoreFlag(valuePtr interface{}) (ok bool, more bool) {
 	v, ok, more := c.receiveAsyncImpl(nil)
-	c.assignValue(v, valuePtr)
+	err:=c.assignValue(v, valuePtr)
+	if err != nil {
+		// failed so serialize corrupt signal. Dropping signal
+		return false, more
```

Comment:
instead of returning here, we should continue go get next signal until there is no more signal.

Reply:
This is in receiveAsync. I think it's not meant to be blocking

Reply:
resolved. Will keep consuming until channel is drained or a valid signal is received